lib/galaxy/jobs/__init__.py

```diff
@@ -1298,11 +1298,12 @@ def path_rewriter( path ):
                             dataset.set_peek( is_multi_byte=True )
                         else:
                             dataset.set_peek()
-                    try:
-                        # set the name if provided by the tool
-                        dataset.name = context['name']
-                    except:
-                        pass
+                    for context_key in ['name', 'info', 'dbkey']:
+                        try:
+                            context_value = context[context_key]
+                            setattr(dataset, context_key, context_value)
+                        except Exception:
```

Comment:
@jmchilton I know this was pre-existing, but is the expected Exception here only `KeyError` ?
Maybe sth. like
```
context_value = context.get(context_key, None)
if context_value:
    setattr(dataset, context_key, context_value)
```
would be better ?